tests/test_apigw.py

```diff
@@ -214,3 +216,24 @@ def test_rest_stage_update(self):
             self.assertEqual(m["loggingLevel"], "INFO")
             self.assertEqual(m["dataTraceEnabled"], True)
         self.assertTrue(found)
+
+    def test_rest_stage_delete(self):
+        session_factory = self.replay_flight_data("test_rest_stage_delete")
+        p = self.load_policy(
+            {
+                "name": "rest-stage-delete",
+                "resource": "rest-stage",
+                "filters": [{"type": "value", "key": "stageName", "value": "delete-test"}],
+                "actions": [{"type": "delete"}],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        client = session_factory().client("apigateway")
+        try:
```

Comment:
rather than do a try/except you can do a self.assertRaises to catch the error, e.g.:

```python
with self.assertRaises(ClientError):
  client.get_stage(...)
```